src/globus_sdk/services/gcs/client.py

```diff
@@ -115,6 +116,71 @@ def connector_id_to_name(connector_id: UUIDLike) -> str | None:
         }
         return connector_dict.get(str(connector_id))
 
+    #
+    # endpoint methods
+    #
+
+    def get_endpoint(
+        self, query_params: dict[str, t.Any] | None = None
+    ) -> UnpackingGCSResponse:
+        """
+        Get the details of the Endpoint that this client is configured to talk to.
+
+        :param query_params: Additional passthrough query parameters
+
+        .. tab-set::
+
+            .. tab-item:: API Info
+
+                ``GET /endpoint``
+
+                .. extdoclink:: Get Endpoint
+                    :ref: openapi_Endpoint/#getEndpoint
+                    :service: gcs
+        """
+        return UnpackingGCSResponse(
+            self.get("/endpoint", query_params=query_params),
+            "endpoint",
+        )
+
+    def update_endpoint(
+        self,
+        endpoint_data: dict[str, t.Any] | EndpointDocument,
+        *,
+        include: list[t.Literal["endpoint"]] | None = None,
```

Comment:
May I suggest that we update this to `list[Literal["endpoint"]] | Literal["endpoint"] | None = None`, and then use [`utils.commajoin`](https://github.com/globus/globus-sdk-python/blob/0b593d848fe88ea7b0e9a1f877ffbd6c4a3315f4/src/globus_sdk/utils.py#L118)?

The reason is a common bug which creeps in with string handling, since strings are iterable.

As currently written, if someone calls `update_endpoint(..., include="endpoint")`, they'll send `?include=e,n,d,p,o,i,n,t`, and then have to cope with the error.

Because my operating assumption is that most users of the SDK are not running type checkers, the likelihood of such a mistake is pretty high, in my experience.

(You can see `get_collection_list` below doing this same kind of handling with `filter`, but in a way which predates the `commajoin` utility.)

Reply:
Sure